mcs/class/System/Test/System.Diagnostics/ProcessTest.cs

```diff
@@ -1130,5 +1131,52 @@ public void HigherPrivilegeProcessName ()
 
 			string v = Process.GetProcessById (1).ProcessName;
 		}
+
+		[Test]
+		public void NonChildProcessWaitForExit ()
+		{
+			if (!RunningOnUnix)
+				Assert.Ignore ("accessing parent pid, only available on unix");
+
+			using (Process process = Process.GetProcessById (getppid ()))
+			{
+				Assert.IsFalse (process.WaitForExit (10), "#1");
+				Assert.IsFalse (process.HasExited, "#2");
+				Assert.Throws<InvalidOperationException>(delegate { int exitCode = process.ExitCode; }, "#3");
+
+				process.Exited += (s, e) => Assert.Fail ("#4");
+
+				Assert.IsFalse (process.WaitForExit (10), "#5");
+				Assert.IsFalse (process.HasExited, "#6");
+				Assert.Throws<InvalidOperationException>(delegate { int exitCode = process.ExitCode; }, "#7");
+			}
+		}
+
+		[Test]
+		public void NonChildProcessName ()
+		{
+			if (!RunningOnUnix)
+				Assert.Ignore ("accessing parent pid, only available on unix");
+
+			using (Process process = Process.GetProcessById (getppid ()))
+			{
+				string name = process.ProcessName;
+			}
+		}
+
+		[Test]
+		public void NonChildProcessId ()
+		{
+			if (!RunningOnUnix)
+				Assert.Ignore ("accessing parent pid, only available on unix");
+
+			using (Process process = Process.GetProcessById (getppid ()))
+			{
+				int id = process.Id;
```

Comment:
maybe `Assert.Greater(id, 0);`, or verify that it's the same as the id returned from getppid()